File: src/bin/place_trade_worker.rs
use dotenv::dotenv;
use std::env;
use nats::asynk::Connection;
use crate::services::nats_service::connect_to_nats;
use crate::services::place_trade_service::run_place_trade_request_worker;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    dotenv().ok();

    let nats_url = env::var("NATS_URL").expect("NATS_URL must be set");
    let nats = connect_to_nats(&nats_url).await?;

    run_place_trade_request_worker(nats).await?;

    Ok(())
}

File: Go/bin/place_trade_worker.go
package main

import (
    "github.com/joho/godotenv"
    "log"
    "os"
    "stock_broker/services"
)

func main() {
    err := godotenv.Load()
    if err != nil {
        log.Fatalf("Error loading .env file")
    }

    natsURL := os.Getenv("NATS_URL")
    nc, err := services.ConnectToNATS(natsURL)
    if err != nil {
        log.Fatalf("Failed to connect to NATS: %v", err)
    }

    services.RunPlaceTradeRequestWorker(nc)
}